svsim/src/main/scala/Simulation.scala

```diff
@@ -367,7 +367,7 @@ object Simulation {
 
   final case class Value(bitCount: Int, asBigInt: BigInt)
 
-  final case class Port private[Simulation] (controller: Simulation.Controller, id: String, info: ModuleInfo.Port) {
+  final case class Port(controller: Simulation.Controller, id: String, info: ModuleInfo.Port) {
```

Comment:
While it's probably fine to make the constructor for `VerilogPreprocessorDefine`, I don't think we want this one to be public. So we should probably add `-Xsource-features: case-apply-copy-access` which will use the Scala 3 behavior (private apply and copy methods) in Scala 2 (this is an API change but it's the desirable behavior).